### PR TITLE
Rewrite api code

### PR DESCRIPTION
**Pull request by TyrenDe**
Removed HttpListener
Add Routes and Controllers

### CODE REVIEW

File: MixItUp.Desktop/Services/WindowsDeveloperAPIService.cs [hunk 1]
```diff
@@ -71,300 +71,321 @@ public UserDeveloperAPIModel(UserDataViewModel userData)
...
+                this.webApp.Dispose();
+                this.webApp = null;
             }
-            await this.CloseConnection(listenerContext, HttpStatusCode.BadRequest, "This is not a valid API");
         }
+    }
 
-        private async Task ProcessDeveloperAPIRequest(HttpListenerContext listenerContext, string httpMethod, List<string> urlSegments, string data)
+    [RoutePrefix("api/mixer/users")]
+    public class MixerController : ApiController
```
Comment (SaviorXTanren) [inline MixItUp.Desktop/Services/WindowsDeveloperAPIService.cs]: Wouldn't this be the MixerUsersController?

File: MixItUp.Desktop/Services/WindowsDeveloperAPIService.cs [hunk 2]
```diff
@@ -71,300 +71,321 @@ public UserDeveloperAPIModel(UserDataViewModel userData)
...
+            allCommands.AddRange(ChannelSession.Settings.EventCommands);
+            allCommands.AddRange(ChannelSession.Settings.TimerCommands);
+            allCommands.AddRange(ChannelSession.Settings.ActionGroupCommands);
+            allCommands.AddRange(ChannelSession.Settings.GameCommands);
+            return allCommands;
+        }
+    }
+
+    [RoutePrefix("api/spotify")]
+    public class SpotifyController : ApiController
```
Comment (SaviorXTanren) [inline MixItUp.Desktop/Services/WindowsDeveloperAPIService.cs]: Let's just get rid of this API for now, we can eventually replace it with Song Requests.

File: MixItUp.Desktop/Services/WindowsDeveloperAPIService.cs (hunk 1 above)
Comment (TyrenDe) [reply MixItUp.Desktop/Services/WindowsDeveloperAPIService.cs]: Updated.

File: MixItUp.Desktop/Services/WindowsDeveloperAPIService.cs (hunk 2 above)
Comment (TyrenDe) [reply MixItUp.Desktop/Services/WindowsDeveloperAPIService.cs]: Removed.

**Pull request merged**